### ISSUE TITLE
How can I get v1-style "lax" string coercion behavior in v2?

### ISSUE THREAD

**Issue by dmontagu** [pydantic/pydantic#5606]
I received this message from a user attempting to migrate to v2:

> I've noticed that integers are no longer coerced into strings when passed to string fields and instead throw an error. How could I change that behavior and is there anywhere I can read about coercion changes?

----

So far, the best approach I'm aware of for reverting to "lax" handling (thanks @Kludex):

```python
from pydantic import BaseModel, field_validator


class User(BaseModel):
    a: str

    @field_validator('a', mode='plain')
    def check_a(cls, v):
        # Could check the type of v before converting, if you only want to allow str or int
        return str(v)
```

Is there any simpler way to achieve this?

----

Regarding docs related to conversion, there will be an updated conversion table in the v2 docs, I'll keep this issue open until someone can add a link to it.

**Comment by josecsotomorales** [comment]
I have a `parameters` field on SQLAlchemy that stores a JSON file with different types, like `dict[str, Any]`, the getter model with: `parameters: dict[str, str]`, so Pydantic V1 would convert all the values to string, can this now be accomplished with a field_validator like this example above?

**Comment by samuelcolvin** [comment]
You could also use `Dict[are, Annotated[str, BeforeValidator(my_func)]]` but that wouldn't give you the "empty -> None" behavior.

**Comment by alexandriabindas** [comment]
I think you can now add `coerce_numbers_to_str=True` to the model_config, which was addressed and merged here [#7508 ](https://github.com/pydantic/pydantic/pull/7508)

**Comment by sydney-runkle** [comment]
Indeed! Based on `coerce_numbers_to_str`, this can be closed as completed.
